File: backend/src/main/java/com/hmdp/entity/Shop.java
package com.hmdp.entity;

import com.baomidou.mybatisplus.annotation.IdType;
import com.baomidou.mybatisplus.annotation.TableField;
import com.baomidou.mybatisplus.annotation.TableId;
import com.baomidou.mybatisplus.annotation.TableName;
import java.math.BigDecimal;
import lombok.Data;
import lombok.EqualsAndHashCode;

@Data
@EqualsAndHashCode(callSuper = true)
@TableName("tb_shop")
public class Shop extends BaseEntity {

    @TableId(type = IdType.AUTO)
    private Long id;
    private String name;
    private Long typeId;
    private String address;
    private BigDecimal x;
    private BigDecimal y;
    private String images;
    private BigDecimal avgPrice;
    private Integer comments;
    private BigDecimal score;
    private String openHours;

    @TableField(exist = false)
    private Double distance;
}

File: frontend/src/api/shop.ts
import { http } from './http'

export interface Shop {
  id: number
  name: string
  typeId: number
  address: string
  images?: string | null
  avgPrice?: number | null
  comments: number
  score: number
  openHours?: string | null
  distance?: number | null
}

export interface ShopType {
  id: number
  name: string
  sort: number
  icon?: string | null
}

export async function fetchShopTypes() {
  const { data } = await http.get('/shops/types')
  return data.data as ShopType[]
}

export async function fetchShopList(typeId?: number, page = 1) {
  const { data } = await http.get('/shops', { params: { typeId, page } })
  return data.data as Shop[]
}

export async function fetchShopDetail(shopId: string | number) {
  const { data } = await http.get(`/shops/${shopId}`)
  return data.data as Shop
}

export async function fetchNearbyShops(typeId: number, x: number, y: number, current = 1) {
  const { data } = await http.get('/shops/nearby', { params: { typeId, x, y, current } })
  return data.data as Shop[]
}
